### DISCUSSION TITLE
Overload resolution

### DISCUSSION THREAD

**Discussion by B1Z0N** [dotnet/roslyn#74535] [Ideas] [answered]
Is there an API that allows to do method resolution? So if I have a `INamedTypeSymbol`, name of a method, it's argument types, type arguments, return type, etc to return the right resolution of a method. Right now I need to specifically find which [`AppendFormatted`](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-10.0/improved-interpolated-strings#:~:text=public%20void%20AppendFormatted%3CT%3E(T%20value)%3B%0A%20%20%20%20%20%20%20%20public,object%3F%20value%2C%20int%20alignment%20%3D%200%2C) will be called in custom interpolated string handler. But for that I'll need to rewrite method overload resolution myself, and that's a bad idea. I know that there are `Binder APIs`, but I can't find them anywhere in Roslyn.

**Comment by CyrusNajmabadi** [comment, marked as answer]
There is not. That's because overload resolution needs more than that. It needs the actual expressions written, which may not have types yet.  

If you need to do overload resolution, generate a new syntax tree with the code you are interested in, and get the semantic model for that to look at.

**Reply by B1Z0N** [reply]
> It needs actual expressions written, which may not have types yet

What do you mean by that? In which case expressions don't have a type? Is it about dynamic?

**Reply by CyrusNajmabadi** [reply]
Passing things like null, default, new(), etc.
